File: LeetCode/remove_duplicates_from_sorted_array_ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(!nums.size())
            return 0;
        
        int aux = 0, count = 1;
        nums[aux++] = nums[0];
        for(int i = 1; i < nums.size(); i++) {
            if(nums[i] == nums[aux-1] && count < 2) {
                nums[aux++] = nums[i];
                count++;
            }
            else if(nums[i] != nums[aux-1]) {
                nums[aux++] = nums[i];
                count = 1;
            }
        }
        
        return aux;
    }
};

File: LeetCode/remove_duplicates_from_sorted_array_ii.go
func removeDuplicates(nums []int) int {
    pos, prev, count := 1, nums[0], 1

    for i := 1; i < len(nums); i++ {
        if prev != nums[i] {
            prev = nums[i]     
            count = 1
        } else {
            count++
        }

        if count <= 2 {
            nums[pos] = nums[i]
            pos++
        }
    }

    return pos
}
